backend/src/oe/service/op-block-header.service.ts

```diff
@@ -0,0 +1,137 @@
+      const blockHash = await bitcoinApi.$getBlockHash(
+        blockHeight.value
+      );
+      const {
+        height,
+        version,
+        previousblockhash,
+        merkle_root,
+        timestamp,
+        difficulty,
+        nonce,
+        chainwork,
+        mediantime,
+      } = await bitcoinApi.$getBlock(blockHash);
+
+      const { totalfee, subsidy } = await bitcoinApi.$getBlockStats(blockHash);
+
+      return {
+        height,
+        version,
+        chainWork: chainwork,
+        medianTime: mediantime,
+        previousBlockHash: previousblockhash,
+        merkleRoot: merkle_root,
+        timestamp,
+        difficulty,
+        nonce,
+        reward: totalfee + subsidy,
+        currentBlockHash: blockHash
+      };
+    } catch (error) {
+      logger.err(
+        `Error while fetching block header ${blockHeight.value}: ${error}`
+      );
+      throw error;
+    }
+  }
+
+  public async $syncOlderBlockHeader(UUID: string, currentTip?: number): Promise<void> {
+    try {
+      logger.debug('Syncing older block headers');
+
+      let { value: currentSyncedBlockHeight } = await opBlockHeaderRepository.$getLatestBlockHeight(UUID);
+
+
+      // Only using fetching current block tip if not present in argument
+      if (!currentTip) {
+        currentTip = await bitcoinApi.$getBlockHeightTip();
+      }
+      const { value: latestConfirmedBlockHeight } = this.verifyConfirmedBlockHeight(currentTip - config.OP_ENERGY.CONFIRMED_BLOCKS_AMOUNT, { value: currentTip });
+
+
+      logger.debug(
+        `currentSyncedBlockHeight: ${currentSyncedBlockHeight}, latestConfirmedBlockHeight: ${latestConfirmedBlockHeight}`
+      );
+
+      if (currentSyncedBlockHeight >= latestConfirmedBlockHeight) {
+        logger.debug('Already synced block headers.');
+        return;
+      }
+
+      currentSyncedBlockHeight += 1;
+
+      logger.debug(
+        `Syncing block header from #${currentSyncedBlockHeight} to #${latestConfirmedBlockHeight}`
+      );
+
+      while (latestConfirmedBlockHeight >= currentSyncedBlockHeight) {
+        const noOfBlockHeaders = Math.max(
+          1,
+          Math.min(10, latestConfirmedBlockHeight - currentSyncedBlockHeight)
+        );
+
+        const blockHeaders = await Bluebird.map(
+          [...Array(noOfBlockHeaders).keys()].map(
+            (i) => i + currentSyncedBlockHeight
+          ),
+          (height) => this.$getBlockHeaderData({ value: height })
+        );
+
+        await Bluebird.mapSeries(blockHeaders, (blockHeader) =>
+          opBlockHeaderRepository.$saveBlockHeaderInDatabase(UUID, blockHeader as BlockHeader)
+        );
+
+        currentSyncedBlockHeight += noOfBlockHeaders;
+      }
+
+      logger.debug('Synced all the missing block headers');
+    } catch (error) {
+      logger.err(`Something went wrong while syncing block header.` + error);
+    }
+  }
+
+  public async $getBlockHeader(UUID: string, height: number): Promise<BlockHeader | null> {
```

Comment:
I am not using this in my code this is for future use if some one wants to get block heder so they can fetch it.

Reply:
well, if you think, that users should use service `OpBlockHeaderService.$getBlockHeader` instead of `opBlockHeaderRepository.$getBlock`, then height should be of type ConfirmedBlockHeight.

I should mention is that every time you are constructing ConfirmedBloockHeight by using `{ value: Something}` instead of using verifyConfirmedBlockHeight() function you are basically defeating the purpose of ConfirmedBlockHeight, which purpose is to make sure, that caller had confirmed that given block height is less or equal tip - 6...

So ideally, there should be no way to create ConfirmedBlockHeight value without verfiyConfirmedBlockHeight() function, but, unfortunately, TypeScript does not allow such granularity. But if you do this, then I expect that there will be some check of value, that is missing....